### PR TITLE
Universal weldable component

### PR DESCRIPTION
**Pull request by Macoron**
## About the PR 
Add universal weldable component for `EntityStorageComponent` and `DoorComponent`. It can probably reused for vents, emitters, etc.

### CODE REVIEW

File: Content.Shared/Storage/SharedStorageComponent.cs [hunk 1]
```diff
@@ -81,8 +81,6 @@ public AnimateInsertingEntitiesEvent(EntityUid storage, List<EntityUid> storedEn
     public enum StorageVisuals
```
Comment (metalgearsloth) [inline Content.Shared/Storage/SharedStorageComponent.cs:81]:
```suggestion
    public enum StorageVisuals : byte
```

File: Content.Server/Tools/Systems/WeldableSystem.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        if (!Resolve(uid, ref component))
+            return false;
+
+        if (!CanWeld(uid, tool, user, component))
+            return false;
+
+        component.BeingWelded = true;
+        _toolSystem.UseTool(tool, user, uid, component.FuelConsumption,
+            component.WeldingTime.Seconds, component.WeldingQuality,
+            new WeldFinishedEvent(user, tool), new WeldCancelledEvent(), uid);
```
Comment (metalgearsloth) [inline Content.Server/Tools/Systems/WeldableSystem.cs:66]: Would be good to have some way for other stuff to specify a modifier if it wants it to take longer.

**Review by metalgearsloth — APPROVED** [review]

File: Content.Server/Tools/Systems/WeldableSystem.cs (hunk 2 above)
Comment (Macoron) [reply Content.Server/Tools/Systems/WeldableSystem.cs:66]: I could probably add a public function in `WeldableSystem` to change welding duration. Adding new event like `WeldingTimeModifierEvent` seems to be overkill for now.

File: Content.Server/Tools/Components/WeldableComponent.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    [DataField("time")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan WeldingTime = TimeSpan.FromSeconds(1f);
+
+    /// <summary>
+    ///     Shown when welded entity is examined.
+    /// </summary>
+    [DataField("weldedExamineMessage")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public string? WeldedExamineMessage = "door-component-examine-is-welded";
```
Comment (ShadowCommander) [inline Content.Server/Tools/Components/WeldableComponent.cs:48]: Localization string should be renamed

**Review by ShadowCommander — CHANGES_REQUESTED** [review]
Construction welded conditions could probably also be converted, but that can be done in a separate PR.

**Pull request merged**